tick/stateful/functions.go

```diff
@@ -624,3 +628,33 @@ func (*humanBytes) Call(args ...interface{}) (v interface{}, err error) {
 	}
 	return
 }
+
+type ifFunc struct {
+}
+
+func (*ifFunc) Reset() {
+
+}
+
+func (*ifFunc) Call(args ...interface{}) (interface{}, error) {
+	if len(args) != 3 {
+		return nil, errors.New("if expects exactly three arguments")
+	}
+
+	var condition bool
+	var isBool bool
+
+	if condition, isBool = args[0].(bool); !isBool {
+		return nil, fmt.Errorf("first argument to if must be a condition with type of bool - got %T", args[0])
+	}
+
+	if reflect.TypeOf(args[1]) != reflect.TypeOf(args[2]) {
+		return nil, fmt.Errorf("Multiple return types isn't supported - second argument is %T and third argument is %T", args[1], args[2])
```

Comment:
nit:

I think this would read better if it read: `Different return types are not supported - ...`